src/pages/collections/[collectionAddress]/activity.tsx

```diff
@@ -0,0 +1,321 @@
+      >
+        <div className={`mx-4 flex w-full items-center justify-between gap-4`}>
+          <div className="text-2xl text-white">Activity</div>
+          <div className="flex justify-end gap-4">
+            <TopLevelFilterButton
+              title={`All events`}
+              onClick={async () => {
+                setSelectedFilter(SelectedFilter.ALL_EVENTS);
+                await refetch({
+                  ...variables,
+                  eventTypes: null,
+                });
+              }}
+              selected={selectedFilter === SelectedFilter.ALL_EVENTS}
+            />
+
+            <TopLevelFilterButton
+              title={`Listings`}
+              onClick={async () => {
+                setSelectedFilter(SelectedFilter.LISTINGS);
+                await refetch({ ...variables, eventTypes: ['LISTINGS'] });
+              }}
+              selected={selectedFilter === SelectedFilter.LISTINGS}
+            />
+
+            <TopLevelFilterButton
+              title={`Offers`}
+              onClick={async () => {
+                setSelectedFilter(SelectedFilter.OFFERS);
+                await refetch({ ...variables, eventTypes: ['OFFERS'] });
+              }}
+              selected={selectedFilter === SelectedFilter.OFFERS}
+            />
+
+            <TopLevelFilterButton
+              title={`Sales`}
+              onClick={async () => {
+                setSelectedFilter(SelectedFilter.SALES);
+                await refetch({ ...variables, eventTypes: ['PURCHASES'] });
+              }}
+              selected={selectedFilter === SelectedFilter.SALES}
+            />
+
+            {/* <TopLevelFilterButton
+              title={`Transfers`}
+              onClick={async () => {
+                setSelectedFilter(SelectedFilter.TRANSFERS);
+                await refetch({ ...variables, eventTypes: ['TRANSFERS'] });
+              }}
+              selected={selectedFilter === SelectedFilter.TRANSFERS}
+            /> */}
+          </div>
+        </div>
+      </div>
+
+      <div className="mt-20 w-full">
+        <NFTGrid
+          showCollection={false}
+          ctaVariant={`collectionActivity`}
+          hasMore={hasMore && activities.length > INITIAL_FETCH - 1}
+          onLoadMore={async (inView: boolean) => {
+            if (!inView || loading || activities.length <= 0) {
+              return;
+            }
+
+            const { data: newData } = await fetchMore({
+              variables: {
+                ...variables,
+                limit: INFINITE_SCROLL_AMOUNT_INCREMENT,
+                offset:
+                  activities.length > INFINITE_SCROLL_AMOUNT_INCREMENT
+                    ? activities.length
+                    : INFINITE_SCROLL_AMOUNT_INCREMENT,
+              },
+              updateQuery: (prev, { fetchMoreResult }) => {
```

Comment:
https://www.apollographql.com/docs/react/pagination/offset-based#the-offsetlimitpagination-helper

Apollo has a helper that handles merging the results. You set it up on the client. Give this a look.